gapidapk/deviceinfo.go

```diff
@@ -211,9 +211,9 @@ func launchPerfettoProducerFromApk(ctx context.Context, d adb.Device, startFunc
 	crash.Go(func() {
 		buf := bufio.NewReader(reader)
 		for {
+			startFuncOnce := task.Once(task.Delay(startFunc, 3*time.Second))
```

Comment:
I'm not sure this is a great solution as it will add a 3 second delay when determining `gpu.counter` support.

I think the original solution of writing to `stderr` is correct. It just could do with being documented somewhere.

However, I think it *should* error out and dump and error message if the producer doesn't write to `stderr` within 3 seconds.

As this in in the `for` loop, won't it keep creating delayed functions if the producer *does* write to `stderr`?

A possible solution would be:

```
startFuncOnce = task.Once(startFunc, 3*time.Second)
startFuncDelayed = task.Delay(func() {
  log.W(ctx, "libgpudataproducer.so did not write to stderr to signify it has started")
  startFuncOnce(ctx)
})
startFuncDelayed()
for {
  line, e := buf.ReadString('\n')
  startFuncOnce(ctx)
  startFuncDelayed.cancel()  // <-- no idea if this exists...